src/core/components/GameCanvas/utils/Border.ts

```diff
@@ -0,0 +1,66 @@
+import { DrawCanvasProps } from '../Canvas.types';
+import { Point } from './Point';
+import { Vector } from './Vector';
+
+export interface IBorder {
+    lines: Vector[];
+    lineColor: string;
+    fillColor: string;
+}
+
+export class Border implements IBorder {
+    lines: Vector[];
+
+    lineColor: string;
+
+    fillColor: string;
+
+    constructor (lineColor: string, fillColor: string) {
+        this.lines = [];
+        this.lineColor = lineColor;
+        this.fillColor = fillColor;
+    }
+
+    addLine (start: Point, end: Point) {
+        const line = new Vector(start, end);
+        this.lines.push(line);
+    }
+
+    collision (startPoint: Point, aroundPoints: Point[]) {
+        for (let i = 0; i < this.lines.length; i += 1) {
+            for (let j = 0; j < aroundPoints.length; j += 1) {
+                const endPoint = aroundPoints[j];
+                const v1 = new Vector(startPoint, endPoint);
+                if (Vector.isIntersection(this.lines[i], v1)) {
+                    return this.lines[i];
+                }
+            }
+        }
+        return null;
+    }
```

Comment:
Предлагаю оптимизировать по памяти

```suggestion
    collision (startPoint: Point, aroundPoints: Point[]) {
        for (let j = 0; j < aroundPoints.length; j += 1) {
            const endPoint = aroundPoints[j];
            const v1 = new Vector(startPoint, endPoint);

            for (let i = 0; i < this.lines.length; i += 1) {
                if (Vector.isIntersection(this.lines[i], v1)) {
                    return this.lines[i];
                }
            }
        }
        return null;
    }
```